pkgs/development/compilers/solc/default.nix

```diff
@@ -10,69 +20,98 @@ assert z3Support -> z3 != null && lib.versionAtLeast z3.version "4.6.0";
+    sha256 = "18230bg4rq9pmm5f8f65j444jpq56rld4fhmpham8q3vr1c1bdjh";
   };
 
-  postPatch = ''
-    substituteInPlace cmake/jsoncpp.cmake \
-      --replace "${jsoncppUrl}" ${jsoncpp}
-  '';
-
-  cmakeFlags = [
-    "-DBoost_USE_STATIC_LIBS=OFF"
-  ] ++ lib.optionals (!z3Support) [
-    "-DUSE_Z3=OFF"
-  ] ++ lib.optionals (!cvc4Support) [
-    "-DUSE_CVC4=OFF"
-  ];
-
-  nativeBuildInputs = [ cmake ];
-  buildInputs = [ boost ]
-    ++ lib.optionals z3Support [ z3 ]
-    ++ lib.optionals cvc4Support [ cvc4 cln gmp ];
-  checkInputs = [ ncurses python3 ];
-
-  # Test fails on darwin for unclear reason
-  doCheck = gccStdenv.hostPlatform.isLinux;
-
-  checkPhase = ''
-    while IFS= read -r -d ''' dir
-    do
-      LD_LIBRARY_PATH=$LD_LIBRARY_PATH''${LD_LIBRARY_PATH:+:}$(pwd)/$dir
-      export LD_LIBRARY_PATH
-    done <   <(find . -type d -print0)
-
-    pushd ..
-    # IPC tests need aleth avaliable, so we disable it
-    sed -i "s/IPC_ENABLED=true/IPC_ENABLED=false\nIPC_FLAGS=\"--no-ipc\"/" ./scripts/tests.sh
-    for i in ./scripts/*.sh ./scripts/*.py ./test/*.sh; do
-      patchShebangs "$i"
-    done
-    TERM=xterm ./scripts/tests.sh
-    popd
-  '';
-
-  meta = with lib; {
-    description = "Compiler for Ethereum smart contract language Solidity";
-    homepage = "https://github.com/ethereum/solidity";
-    license = licenses.gpl3;
-    platforms = with platforms; linux ++ darwin;
-    maintainers = with maintainers; [ dbrock akru lionello sifmelcara ];
-    inherit version;
+  solc = gccStdenv.mkDerivation rec {
+    pname = "solc";
+    version = "0.8.2";
+
+    # upstream suggests avoid using archive generated by github
+    src = fetchzip {
+      url = "https://github.com/ethereum/solidity/releases/download/v${version}/solidity_${version}.tar.gz";
+      sha256 = "11w7sa1y2dirzh84k04fkwbfc6xpjp5jr65w1pmb2pnkjvvf46xq";
+    };
+
+    patches = [
+      # not yet merged, but required for build
+      # see https://github.com/ethereum/solidity/pull/11045
+      (fetchpatch {
+        name = "ensure-no-_static-suffix.patch";
+        url = "https://github.com/ethereum/solidity/commit/5928b9261323f1e0c1a480bf36861a0657b34c72.patch";
+        sha256 = "16py0992jdrwgign3sc79aj9ijingkdcaavzsc2ibpicxj2h2scq";
+      })
+    ];
+    postPatch = ''
+      substituteInPlace cmake/jsoncpp.cmake \
+        --replace "${jsoncppUrl}" ${jsoncpp}
+      substituteInPlace cmake/range-v3.cmake \
+        --replace "${range3Url}" ${range3}
+    '';
+
+    cmakeFlags = [
+      "-DBoost_USE_STATIC_LIBS=OFF"
+    ] ++ lib.optionals (!z3Support) [
+      "-DUSE_Z3=OFF"
+    ] ++ lib.optionals (!cvc4Support) [
+      "-DUSE_CVC4=OFF"
+    ];
+
+    nativeBuildInputs = [ cmake ];
+    buildInputs = [ boost ]
+      ++ lib.optionals z3Support [ z3 ]
+      ++ lib.optionals cvc4Support [ cvc4 cln gmp ];
+    checkInputs = [ ncurses python3 ];
+
+    # tests take 60+ minutes to complete, only run as part of passthru tests
+    doCheck = false;
```

Comment:
nit: Because solc is kind of security-critical. I'm not sure if it is a good idea to disable tests (especially when nix has some uncommon build behavior). Is there a compromised way to keep the test?

Reply:
I think that 30+ mins is unacceptable for building a package. It puts way more strain on resources to be able to validates for long periods of time. Especially in cases of an overlay causing rebuilds, no one wants to sit through 2-4 hours of tests to be conducted. I would be okay with running a subset of tests, but I'm not sure of what is available, only documentation I could find is https://docs.soliditylang.org/en/latest/contributing.html#running-the-tests and I'm not super familiar with the project. I just have an interest in being able to compile solidity 0.8 dependent code.

Also, the ownership of security more on upstream than downstream. Really, I'm just trying to verify it's working as intended.

Reply:
Ack